account_payment_order/models/account_payment_order.py

```diff
@@ -141,6 +141,10 @@ def create(self, vals):
             vals['payment_type'] = payment_mode.payment_type
             if payment_mode.bank_account_link == 'fixed':
                 vals['journal_id'] = payment_mode.fixed_journal_id.id
+            if (
+                    'date_prefered' not in vals and
```

Comment:
The first term of the condition could be written as:
`not vals.get('date_prefered', None) and`

It's true that on a create, usually invoked saving a new record by user interface, fields set to False / None are not even passed as arguments (and anyway, `date_prefered` is defined as required and with a default, so when invoked by user interface it will never happen to not find it in the arguments). Still, if the create is invoked differently, like from inside another method or through an external XML-RPC request, it could easily happen that `'date_prefered'` is in `vals` but it's set to _None / False_. The suggested change would correctly address this case.